### ISSUE TITLE
Error page always rendered in default language for some languages

### ISSUE THREAD

**Issue by afbora** [getkirby/kirby#4834]
### Scenario to reproduce:

Have 2 languages:
- French (default), with `"url" => "/"`
- English (secondary), with `"url" => "en"`

If you visit `/en/asdf`, the url is right, but the content is not translated (appears in default language).
If you visite `/asdf`, the url is right and the content is translated.

Fix:
If I set the french `url => "fr"`, it works as intended. Both "/" and "/en" are displaying the right locale.
If I set the english as default, with `url => "/"`, it works as intended.

This means I cannot have French as default language with `url => "/"`.

Same issue as this user: https://forum.getkirby.com/t/error-page-is-shown-in-wrong-language/18084/9

_Originally posted by @juliencharette in https://github.com/getkirby/kirby/issues/2609#issuecomment-1306071201_

**Comment by afbora** [comment]
This issue is only reproducible for some languages, we've tested following languages:

- FR ❌ 
- IT ❌ 
- TR ❌ 
- ES ❌ 
- DE ✅ 
- RU ❌
- NL ❌

**Comment by adamkiss** [comment]
Ha, it almost looks like only `$language <= "en"` can have a root url.

**Comment by lukasbestle** [comment]
That's super weird...

**Comment by distantnative** [comment]
> Ha, it almost looks like only $language <= "en" can have a root url.

Is it just `en` or is it `en` and `de` from your test, @afbora @adamkiss?

**Comment by adamkiss** [comment]
@distantnative I'm not involved, I just noticed it looks like it's broken alphabetically.

**Comment by afbora** [comment]
@distantnative EN is always the secondary language in manual tests. Only DE passed the test when we changed the primary language.

**Comment by distantnative** [comment]
I think I am getting to the source of the problem, without quite knowing yet how to solve it:

- Example: `http://sandbox.test/en/not-there`
- In a multilang setup, this is where we set up the routes for each language https://github.com/getkirby/kirby/blob/main/src/Cms/LanguageRoutes.php#L31
- if nothing can be found for the language, we trigger `$this->next()` to jump to fallback routes https://github.com/getkirby/kirby/blob/main/src/Cms/LanguageRoutes.php#L47
- I think this is where it happens that the matched path `not-there` for the secondary language English does not find a page and falls back to looking for the path `en/not-there` in the French primary language. Which also can't be found - but since we are looking now in the French context, the French error page gets rendered.

On the one hand, I see how we thought a fallback to also check for a page `en/not-there` in the `/` context would make sense. But it doesn't quite when we already checked for `not-there` in the `/en` context. It creates parallel universes :D

This is why right now the secondary-language error page cannot really be reached. I didn't check for German, but I would be surprised if that one is actually working either.

**Comment by distantnative** [comment]
@afbora I tried to tackle this again, but cannot reproduce it currently with `develop-minor`. Can you still?

**Comment by distantnative** [comment]
I looked into this again. I am not sure we can solve this:

- The issue occurs when the default language has no own prefix, but `/`
- Our language routes are set up that it tries to match the path in a route for each language and move `next` if nothing is found (e.g. also to support custom routes that aren't set up by LanguageRoutes but share the prefix)
- Since we move to the next routes, a path like `en/foo` matched for the secondary language English, didn't find the page, moves on and now matches the pattern for the primary language French route because it has no prefix. Now as `en/foo` full path. And thus the language is set to French and if no page is found, it's the French error page.

I see two options, both not sure how to implement yet:
a) We implement a way where we don't move to the next route, but skip all other language routes while not skipping all other routes.
b) We find a way how a language without URL prefix cannot match any path that would match any other language.

**Comment by afbora** [comment]
I've tested again on old Kirby version 3.3.5 and still reproducible. 
Now I've tested on latest Kirby version as 5.3.3, issue is gone and works as exptected 🎉

**Comment by wdebusschere** [comment]
We still have this problem in 5.5.3
@afbora @distantnative

**Comment by afbora** [comment]
@wdebusschere I was checked before. Let me test again.

**Comment by afbora** [comment]
@wdebusschere You are right, I tested the wrong combination back then and assumed it was fixed. Sorry for the noise.

**Comment by afbora** [comment]
@distantnative I have an idea for a fix: leave the route patterns untouched and, in `LanguageRoutes`, restore the language whose prefix actually matched the URL whenever a language without URL prefix cannot resolve the path. Route order no longer matters and the fallback chain stays intact.

Should I open a PR so you can take a look?

**Comment by distantnative** [comment]
@afbora I can't fully follow just by reading the post alone, I think a PR would indeed best to understand and discuss it.

**Comment by afbora** [comment]
@distantnative Here the draft PR https://github.com/getkirby/kirby/pull/8484
